--- a/net/minecraft/data/BuiltinRegistries.java
+++ b/net/minecraft/data/BuiltinRegistries.java
@@ -81,7 +81,7 @@
       f_123867_.put(resourcelocation, () -> {
          return p_235999_.m_236014_(p_235998_);
       });
-      f_123868_.m_203505_(p_235997_, p_235998_, p_236000_);
+      f_123868_.m_203505_((ResourceKey<WritableRegistry<?>>) p_235997_, p_235998_, p_236000_);
       return p_235998_;
    }
 
@@ -92,7 +92,7 @@
    }
 
    public static <V extends T, T> Holder<V> m_206380_(Registry<T> p_206381_, String p_206382_, V p_206383_) {
-      return m_206388_(p_206381_, new ResourceLocation(p_206382_), (T)p_206383_);
+      return (Holder<V>) m_206388_(p_206381_, new ResourceLocation(p_206382_), (T)p_206383_);
    }
 
    public static <T> Holder<T> m_206396_(Registry<T> p_206397_, String p_206398_, T p_206399_) {
