--- a/com/mojang/realmsclient/gui/screens/RealmsSelectWorldTemplateScreen.java
+++ b/com/mojang/realmsclient/gui/screens/RealmsSelectWorldTemplateScreen.java
@@ -15,6 +15,7 @@
 import java.util.Arrays;
 import java.util.Collections;
 import java.util.List;
+import java.util.concurrent.atomic.AtomicReference;
 import java.util.function.Consumer;
 import java.util.stream.Collectors;
 import javax.annotation.Nullable;
@@ -198,17 +199,17 @@
          public void run() {
             WorldTemplatePaginatedList worldtemplatepaginatedlist = p_89654_;
 
-            Either<WorldTemplatePaginatedList, String> either;
+            final AtomicReference<Either<WorldTemplatePaginatedList, String>> either = new AtomicReference<>();
             for(RealmsClient realmsclient = RealmsClient.m_87169_(); worldtemplatepaginatedlist != null; worldtemplatepaginatedlist = RealmsSelectWorldTemplateScreen.this.f_96541_.m_18691_(() -> {
-               if (either.right().isPresent()) {
-                  RealmsSelectWorldTemplateScreen.f_89605_.error("Couldn't fetch templates: {}", either.right().get());
+               if (either.get().right().isPresent()) {
+                  RealmsSelectWorldTemplateScreen.f_89605_.error("Couldn't fetch templates: {}", either.get().right().get());
                   if (RealmsSelectWorldTemplateScreen.this.f_89612_.m_89817_()) {
                      RealmsSelectWorldTemplateScreen.this.f_89604_ = TextRenderingUtils.m_90256_(I18n.m_118938_("mco.template.select.failure"));
                   }
 
                   return null;
                } else {
-                  WorldTemplatePaginatedList worldtemplatepaginatedlist1 = either.left().get();
+                  WorldTemplatePaginatedList worldtemplatepaginatedlist1 = either.get().left().get();
 
                   for(WorldTemplate worldtemplate : worldtemplatepaginatedlist1.f_87753_) {
                      RealmsSelectWorldTemplateScreen.this.f_89612_.m_89804_(worldtemplate);
@@ -227,7 +228,7 @@
                   }
                }
             }).join()) {
-               either = RealmsSelectWorldTemplateScreen.this.m_89655_(worldtemplatepaginatedlist, realmsclient);
+               either.set(RealmsSelectWorldTemplateScreen.this.m_89655_(worldtemplatepaginatedlist, realmsclient));
             }
 
          }
