--- a/net/minecraft/world/entity/animal/camel/Camel.java
+++ b/net/minecraft/world/entity/animal/camel/Camel.java
@@ -127,7 +127,7 @@
 
    protected void m_8024_() {
       this.f_19853_.m_46473_().m_6180_("camelBrain");
-      Brain<?> brain = this.m_6274_();
+      Brain<? super Camel> brain = (Brain<? super Camel>) this.m_6274_();
       brain.m_21865_((ServerLevel)this.f_19853_, this);
       this.f_19853_.m_46473_().m_7238_();
       this.f_19853_.m_46473_().m_6180_("camelActivityUpdate");
